src/Raven.Server/Documents/Handlers/BatchRequestParser.cs

```diff
@@ -950,46 +951,36 @@ private static unsafe CommandType GetCommandType(JsonParserState state, JsonOper
                         return CommandType.AttachmentMOVE;
-
-                    ThrowInvalidProperty(state, ctx);
-                    return CommandType.None;
+                    break;
                 
                 case 16:
                     if (*(long*)state.StringBuffer == 7308612546338255937 &&
                         *(long*)(state.StringBuffer + sizeof(long)) == 4995694080542667886)
                         return CommandType.AttachmentDELETE;
-                    ThrowInvalidProperty(state, ctx);
-                    return CommandType.None;
-
+                    break;
+                
                 case 18:
-                    if (*(long*)state.StringBuffer != 5000528724088418115 ||
-                        *(long*)(state.StringBuffer + sizeof(long)) != 5793150219460305784 ||
-                        *(short*)(state.StringBuffer + sizeof(long) + sizeof(long)) != 21589)
-                    {
-                        ThrowInvalidProperty(state, ctx);
-                    }
-                    return CommandType.CompareExchangePUT;
-              
+                    if (*(long*)state.StringBuffer == 5000528724088418115 &&
+                        *(long*)(state.StringBuffer + sizeof(long)) == 5793150219460305784 &&
+                        *(short*)(state.StringBuffer + sizeof(long) + sizeof(long)) == 21589) 
+                        return CommandType.CompareExchangePUT;
+                    break;
+                
                 case 21:
                     if (*(long*)state.StringBuffer == 5000528724088418115 &&
                         *(long*)(state.StringBuffer + sizeof(long)) == 4928459091005170552 &&
                         *(int*)(state.StringBuffer + sizeof(long) + sizeof(long)) == 1413827653 &&
                         state.StringBuffer[sizeof(long) + sizeof(long) + sizeof(int)] == (byte)'E')
-                       return CommandType.CompareExchangeDELETE;
+                        return CommandType.CompareExchangeDELETE;
                     
                     if (*(long*)state.StringBuffer == 8531315664536891206 &&
                         *(long*)(state.StringBuffer + sizeof(long)) == 7309979286770381673 &&
                         *(int*)(state.StringBuffer + sizeof(long) + sizeof(long)) == 1869182049 &&
                         state.StringBuffer[sizeof(long) + sizeof(long) + sizeof(int)] == (byte)'n')
                         return CommandType.ForceRevisionCreation;
-                    
-                    ThrowInvalidProperty(state, ctx);
-                    return CommandType.None;
-                
-                default:
-                    ThrowInvalidProperty(state, ctx);
-                    return CommandType.None;
+                    break;
             }
+            throw new InvalidCommandTypeException("Invalid command type: " + ctx.AllocateStringValue(null, state.StringBuffer, state.StringSize));
```

Comment:
Better to move it to a dedicated method - like we did with `ThrowInvalidProperty`. The reason is slightly better performance - parsing is our hot path.